### PR TITLE
ref(starlette): Use new scopes API

### PR DESCRIPTION
**Pull request by sentrivana**
---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/integrations/starlette.py [hunk 1]
```diff
@@ -104,58 +104,54 @@ def _enable_span_for_middleware(middleware_class):
...
-            # Update transaction name with middleware name
-            name, source = _get_transaction_from_middleware(app, scope, integration)
-            if name is not None:
-                Scope.get_current_scope().set_transaction_name(
-                    name,
-                    source=source,
-                )
+        integration = sentry_sdk.get_client().get_integration(StarletteIntegration)
+        if integration is None:
+            return await old_call(app, scope, receive, send, **kwargs)
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlette.py:109]: We could use `ensure_integration_enabled_async` here

File: sentry_sdk/integrations/starlette.py [hunk 2]
```diff
@@ -104,58 +104,54 @@ def _enable_span_for_middleware(middleware_class):
...
-
-                return await old_call(app, scope, new_receive, new_send, **kwargs)
+        middleware_name = app.__class__.__name__
 
-        else:
-            return await old_call(app, scope, receive, send, **kwargs)
+        # Update transaction name with middleware name
+        name, source = _get_transaction_from_middleware(app, scope, integration)
+        if name is not None:
+            Scope.get_current_scope().set_transaction_name(
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlette.py:116]: 💯 (right scope used!)

File: sentry_sdk/integrations/starlette.py [hunk 3]
```diff
@@ -389,8 +384,9 @@ def _sentry_request_response(func):
 
             async def _sentry_async_func(*args, **kwargs):
                 # type: (*Any, **Any) -> Any
-                hub = Hub.current
-                integration = hub.get_integration(StarletteIntegration)
+                integration = sentry_sdk.get_client().get_integration(
+                    StarletteIntegration
+                )
                 if integration is None:
                     return await old_func(*args, **kwargs)
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlette.py:391]: We could used `ensure_integration_enabled_async` here

File: sentry_sdk/integrations/starlette.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlette.py:109]: I was using it initially but it seems there's something wrong with it. When I [removed](https://github.com/getsentry/sentry-python/pull/2877/commits/25fdff1ad903942ada1a69d4e94c59b4a51da65a) it the tests started passing. So we should probably fix the async decorator. (The sync one works fine.)

File: sentry_sdk/integrations/starlette.py (hunk 3 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlette.py:391]: See above. Will open an issue for this.

File: sentry_sdk/integrations/starlette.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlette.py:109]: https://github.com/getsentry/sentry-python/issues/2892

Comment (antonpirker) [reply sentry_sdk/integrations/starlette.py:109]: Ok, I guess than it is ok.

File: sentry_sdk/integrations/starlette.py (hunk 3 above)
Comment (antonpirker) [reply sentry_sdk/integrations/starlette.py:391]: see answer above :-)

**Review by antonpirker — APPROVED** [review]

**Pull request merged**